runtime/src/bank.rs

```diff
@@ -352,6 +352,9 @@ pub struct Bank {
     /// Rewards that were paid out immediately after this bank was created
     #[serde(skip)]
     pub rewards: Option<Vec<(Pubkey, i64)>>,
+
+    #[serde(skip)]
+    pub skip_drop: AtomicBool,
```

Comment:
After dropping the old version of the bank in the purge logic here: https://github.com/solana-labs/solana/pull/9805/files#diff-5984b6b0429f857c13a2a362669ab37cR520-R522, I was surprised to see that on replaying the new version of the slot, accounts were randomly getting dropped.

Turns out the Drop implementation of the old bank was running a little later, purging the slot accounts. Not entirely sure where that extra reference may be hiding, but I thought it best not to assume there were no outstanding references.

Reply:
I'm afraid but I rather want to avoid these special exceptions. That makes behavior reasoning very hard quickly. And those flags tend to be abused in general once after introduced, IMO. ;) For example, this is how troublesome to remove a special exception around something called `last_root` after added in haste: https://github.com/solana-labs/solana/pull/8148/files#r378052963 

I feel it's bothersome enough, but I to want know whether there is possibility to avoid this or not by finding a way to correctly `Drop`-ping in slot-descending order ({child,newer}-first). If this is the last resort with good confidence, I'm completely fine to do so. :)

And here is [my bet](https://github.com/solana-labs/solana/pull/9805/files/70a2dd47c0cb8e7d9d20d4d29928643115b5b880#r418523160)

Reply:
@ryoqun good find! However, I added that flag because we don't enforce anywhere that there are no outstanding references to that bank. `BankForks` is passed around in almost all the modules, so it's hard to enforce, even in tests, that no other thread has a reference to the bank. I think it's ugly, but ensures we don't shoot ourselves in the future.

Reply:
ok, i see. Then, let's do the flag thing. :)